crates/testing/src/script.rs

```diff
@@ -107,11 +120,44 @@ pub async fn run_test_script<TYPES, S: TaskState<Event = Arc<HotShotEvent<TYPES>
         }
 
         for assert in &stage.outputs {
-            if let Ok(received_output) = test_receiver.try_recv() {
-                tracing::debug!("Test received: {:?}", received_output);
-                validate_output_or_panic(stage_number, &received_output, assert);
-            } else {
-                panic_missing_output(stage_number, assert);
+            match assert {
```

Comment:
Having to add Left and Right to all of the checks seems like it'd grow to become cumbersome over time, and the Left and Right don't really document what they're doing. What do you think about having a flag like `unordered` or something that just checks outputs without worrying about their order of arrival?

Reply:
I don’t like the use of `either` either 😅, but haven’t a found a better way yet. The problem with using a flag is that we can’t just say event A is unordered. We have to specify that event A and B can be any order, i.e., somehow group them together, so I figured we need to support an `either` type.

Reply:
I feel like a more general way might be to make the predicates return `pub enum PredicateState { Pass, Fail, Incomplete }` instead of a `bool`, and instead of a single `try_recv()` we loop until it passes or fails. We could then make an `all` predicate that just stores events in some internal `HashMap` or something and passes or fails when it's seen them all or gets an unexpected event. 

But we probably don't need to do this now. For now we could just change the `exact` and other predicates in `predicates.rs` to return `Left` or `Right` as appropriate, and then we won't have to write `Left` or `Right` everywhere in the actual tests (and it'll be easier to change things later).

Reply:
@ss-es Both ways sound good! I'll choose the latter one for this PR since it needs fewer changes, but I do like the more general approach and we can probably do it later when needed.

Reply:
I ended up choosing an approach combining these two ideas, with an enum supporting checking two consecutive events.